contracts/Inventory.sol

```diff
@@ -0,0 +1,278 @@
+// SPDX-License-Identifier: MIT
+
+/**
+ * Authors: Omar Garcia<ogarciarevett>
+ * GitHub: https://github.com/lootledger/inventory
+ */
+
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+import "./interfaces/IInventory.sol";
+
+
+contract Inventory is ERC721Holder, ERC1155Holder, ReentrancyGuard, AccessControl, IInventory {
+
+    uint256 constant ERC20_ITEM_TYPE = 20;
+    uint256 constant ERC721_ITEM_TYPE = 721;
+    uint256 constant ERC1155_ITEM_TYPE = 1155;
+
+    bytes32 public constant INVENTORY_ADMIN = keccak256("INVENTORY_ADMIN");
+
+    address public contractERC721Address;
+    uint256 public numSlots;
+
+    // SlotId => slot, useful to get the rest of the slot data.
+    mapping(uint256 => LibInventory.Slot) private slotData;
+
+    // SlotType => "slot type name"
+    mapping(uint256 => string) private slotTypes;
```

Comment:
Is slot type deprecated? I noticed it's not there within any of the InventoryFacets

Reply:
Yes, the slot type will be an "extension" actually this shouldn't be here, thanks for the comment.